drivers/dma/dma_mcux_lpc.c

```
@@ -188,10 +194,66 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 
 	LOG_DBG("channel is %d", p_handle->channel);
 
-	if (config->source_chaining_en && config->dest_chaining_en) {
-		LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
-		/* Link DMA_OTRIG 0 to channel */
-		INPUTMUX_AttachSignal(INPUTMUX, 0, config->linked_channel);
+	if (!config->source_chaining_en && !config->dest_chaining_en) {
+		/* IFF channel is connected to OTrig Mux Then Disconnect */
+		if (!k_sem_take(&is_otrig_being_configured, K_MSEC(200))) {
+			for (int i = 0; i <  dev_config->num_of_otrigs; i++) {
+				if (dma_data->linked_otrig_ch[i] != -1 &&
+				   (config->linked_channel == dma_data->linked_otrig_ch[i] ||
+				    channel == dma_data->linked_otrig_ch[i])) {
+					dma_data->linked_otrig_ch[i] = -1;
+					break;
+				}
+			}
+			k_sem_give(&is_otrig_being_configured);
+		} else { /* k_sem_take(..); */
+			LOG_ERR("DMA attempted to configure an Otrig mux"
+				"while the mux was in use.");
+			return -EACCES;
+		}
+	} else {
+		if (!k_sem_take(&is_otrig_being_configured, K_MSEC(200))) {
+			LOG_DBG("link dma 0 channel %d with channel %d",
+				channel, config->linked_channel);
+			uint8_t is_otrig_available = 0;
+			uint8_t otrig_index = 0;
+
+			for (; otrig_index < dev_config->num_of_otrigs; ++otrig_index) {
+				if (dma_data->linked_otrig_ch[otrig_index] == -1) {
+					is_otrig_available = 1;
+					break;
+				}
+			}
+			if (!is_otrig_available) {
+				LOG_ERR("Calling function tried to setup up otrig #%d"
+				"with at least two sets of channel, each otrig can"
+				"can only support one pair of channels.", otrig_index);
+				return -EINVAL;
+			}
+
+			/* Since INPUTMUX handles the signal for
+			 * channel chaining, we should request that
+			 * the trigger is handled via INPUTMUX hardware
+			 * trigger by default
+			 */
+			DEV_BASE(dev)->CHANNEL[config->linked_channel].CFG |=
+				DMA_CHANNEL_CFG_HWTRIGEN_MASK;
+			DMA_EnableChannel(DEV_BASE(dev), config->linked_channel);
```

instead of enabling the channel, in case it hasnt been configured by this driver, maybe you should error out if the linked channel is not set up in this driver

---

This is the reason for the "For loop" above this statement.

---

I don't see what you mean? What I mean is if a user calls dma_config for channel A and tries to link it to channel B, if they never called dma_config on channel B then the config call for channel A maybe should error since it can caused an unexpected behavior where a channel is triggered that isnt set up

---

I changed my mind about this, I guess it would just add more tracking data to the driver and is already outside the contract of the API, so we can leave it to user responsibility